components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManager.java

```diff
@@ -5187,6 +5187,23 @@ public void updatePermissionListOfGroup(String groupName, String[] permissionToA
         }
     }
 
+    @Override
+    public List<Attribute> getCoreSchema() throws CharonException {
+
+        Map<ExternalClaim, LocalClaim> scimClaimToLocalClaimMap =
+                getMappedLocalClaimsForDialect(SCIMCommonConstants.SCIM_CORE_CLAIM_DIALECT, tenantDomain);
+        Map<String, Attribute> filteredFlatAttributeMap = getFilteredUserSchemaAttributes(scimClaimToLocalClaimMap);
```

Comment:
I feel that `getFilteredUserSchemaAttributes` method is not relevant to this logic. Shall we use `getFilteredSchemaAttributes` method because here we need to filter out local claims mapped to core schema which are marked as  supported by default